feature_engine/encoding/similarity_encoder.py

```diff
@@ -174,9 +182,19 @@ def __init__(
                 "missing_values should be one of 'raise', 'impute' or 'ignore'."
                 f" Got {missing_values!r} instead."
             )
+        if keywords and not isinstance(keywords, dict):
+            raise ValueError(
+                f"keywords should be dict or None. Got {keywords!r} instead."
+            )
+        if keywords and not all(isinstance(item, list) for item in keywords.values()):
```

Comment:
This may be a bit picky, but, should we not test that all items in the lists are strings as well?

Reply:
It does not matter, they will all be converted to strings later in `gpm` function